rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -119,44 +135,151 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
 
   // Get the current simulation time.
   common::Time now = world_->GetSimTime();
-
-  // Calculate the wind force.
-  double wind_strength = wind_force_mean_;
-  math::Vector3 wind = wind_strength * wind_direction_;
-  // Apply a force from the constant wind to the link.
-  link_->AddForceAtRelativePosition(wind, xyz_offset_);
-
-  math::Vector3 wind_gust(0, 0, 0);
-  // Calculate the wind gust force.
-  if (now >= wind_gust_start_ && now < wind_gust_end_) {
-    double wind_gust_strength = wind_gust_force_mean_;
-    wind_gust = wind_gust_strength * wind_gust_direction_;
-    // Apply a force from the wind gust to the link.
-    link_->AddForceAtRelativePosition(wind_gust, xyz_offset_);
+  
+  math::Vector3 wind_velocity(0, 0, 0);
+
+  // Choose user-specified method for calculating wind velocity.
+  if (!custom_static_wind_field_) {
+    // Calculate the wind force.
+    double wind_strength = wind_force_mean_;
+    math::Vector3 wind = wind_strength * wind_direction_;
+    // Apply a force from the constant wind to the link.
+    link_->AddForceAtRelativePosition(wind, xyz_offset_);
+
+    math::Vector3 wind_gust(0, 0, 0);
+    // Calculate the wind gust force.
+    if (now >= wind_gust_start_ && now < wind_gust_end_) {
+      double wind_gust_strength = wind_gust_force_mean_;
+      wind_gust = wind_gust_strength * wind_gust_direction_;
+      // Apply a force from the wind gust to the link.
+      link_->AddForceAtRelativePosition(wind_gust, xyz_offset_);
+    }
+
+    wrench_stamped_msg_.mutable_header()->set_frame_id(frame_id_);
+    wrench_stamped_msg_.mutable_header()->mutable_stamp()->set_sec(now.sec);
+    wrench_stamped_msg_.mutable_header()->mutable_stamp()->set_nsec(now.nsec);
+
+    wrench_stamped_msg_.mutable_wrench()->mutable_force()->set_x(wind.x +
+                                                                 wind_gust.x);
+    wrench_stamped_msg_.mutable_wrench()->mutable_force()->set_y(wind.y +
+                                                                 wind_gust.y);
+    wrench_stamped_msg_.mutable_wrench()->mutable_force()->set_z(wind.z +
+                                                                 wind_gust.z);
+
+    // No torque due to wind, set x,y and z to 0.
+    wrench_stamped_msg_.mutable_wrench()->mutable_torque()->set_x(0);
+    wrench_stamped_msg_.mutable_wrench()->mutable_torque()->set_y(0);
+    wrench_stamped_msg_.mutable_wrench()->mutable_torque()->set_z(0);
+
+    wind_force_pub_->Publish(wrench_stamped_msg_);
+
+    // Calculate the wind speed.
+    double wind_speed = wind_speed_mean_;
```

Comment:
This assignment was present in the original wind plugin, probably to allow for adding a random component to the wind speed using wind_speed_variance_ (which can be defined by the user), although this is currently not implemented. 
But this could be done directly in the following line anyways. I therefore deleted this assignment.